dotCMS/src/integration-test/java/com/dotmarketing/servlets/BinaryExporterServletTest.java

```diff
@@ -0,0 +1,209 @@
+                { BY_INODE, NO_PERMISSIONS },
+        };
+    }
+
+
+    @Test
+    @UseDataProvider("testCases")
+    public void requestBinaryFile(
+            final String byIdType, final String permissionType) throws Exception {
+
+        final boolean byIdentifier = byIdType.equals(BY_ID);
+        final boolean permissionsRequired = permissionType.equals(NO_PERMISSIONS);
+
+        Contentlet fileAsset = null;
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+        try (TmpBinaryFile tmpSourceFile = new TmpBinaryFile(true);
+             TmpBinaryFile tmpTargetFile = new TmpBinaryFile(false)) {
+
+            // Checkin file
+            fileAsset = checkinFileAsset(tmpSourceFile, folder);
+
+            // Set asset permissions
+            if (permissionsRequired) {
+                addPermissions(fileAsset);
+            }
+
+            // Build request and response
+            final String fileURI = "/contentAsset/raw-data/"
+                    + (byIdentifier ? fileAsset.getIdentifier() : fileAsset.getInode())
+                    + "/fileAsset/";
+            final HttpServletRequest request = mockServletRequest(fileURI);
+            final HttpServletResponse response = mockServletResponse(tmpTargetFile);
+
+            // Send servlet request
+            sendRequest(request, response);
+
+            if (permissionsRequired) {
+                // Verify response status
+                assertEquals(HttpServletResponse.SC_UNAUTHORIZED, response.getStatus());
+            } else {
+                // Verify response
+                final byte[] responseContent = Files.readAllBytes(tmpTargetFile.getPath());
+                assertArrayEquals(ShortyServletAndTitleImageTest.pngPixel, responseContent);
+            }
+
+        } finally {
+            if (UtilMethods.isSet(fileAsset) && UtilMethods.isSet(fileAsset.getInode())) {
+                FileAssetDataGen.remove(fileAsset);
+            }
+            if (UtilMethods.isSet(folder) && UtilMethods.isSet(folder.getInode())) {
+                FolderDataGen.remove(folder);
+            }
+        }
+
+    }
+
+    private Contentlet checkinFileAsset(final TmpBinaryFile tmpSourceFile, final Folder folder)
+            throws DotSecurityException, DotDataException {
+
+        final Contentlet fileAsset = new FileAssetDataGen(
+                folder, tmpSourceFile.getFile()).nextPersisted();
+        ContentletDataGen.publish(fileAsset);
+        assertTrue(APILocator.getContentletAPI().isInodeIndexed(
+                fileAsset.getInode(), true, 1000));
+        return fileAsset;
+
+    }
+
+    private HttpServletRequest mockServletRequest(final String fileURI) {
+        return new MockSessionRequest(new MockServletPathRequest(
+                new MockHttpRequest("localhost", fileURI).request(),
+                "/contentAsset"));
+    }
+
+    private HttpServletResponse mockServletResponse(TmpBinaryFile tmpTargetFile) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'tmpTargetFile' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885343)